clientconn.go

```diff
@@ -294,6 +294,12 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (*Clien
 	if ok {
 		go cc.lbWatcher()
 	}
+	// Start a goroutine to monitor cc.ctx.
+	// cc.Close() needs to be called if cc.ctx is canceled.
+	go func() {
+		<-cc.ctx.Done()
```

Comment:
What if this is deadline exceeded instead of canceled? It seems we need to refine the DialContext behavior better.